runtime/v2/shim.go

```diff
@@ -227,8 +213,15 @@ func (s *shim) ID() string {
 }
 
 // PID of the task
-func (s *shim) PID() uint32 {
-	return uint32(s.taskPid)
+func (s *shim) PID(ctx context.Context) (uint32, error) {
```

Comment:
Adding `Connect` in the `PID` will cause an additional remote call  in `Wait` and `local.Create`.

just my little opinion.

Reply:
Technically it's not required for the `Wait` call itself, but just to keep response from `containerd` backward compatible.
This changes are mainly done to make shim unaware of the underlying services it offers. We can move taskpid handling to the task service once shim service lands, so there will no need to do this extra call.